### ISSUE TITLE
Exploitability Calculation - ICM

### ISSUE THREAD

**Issue by Barcode9** [b-inary/postflop-solver#28]
Hi, 

first of all very nice and clean project, the GUI in the other project is very impressive. 

I have been working on trying to implement custom ICM payouts, which seems easy enough if I just change the evaluate function to call my custom ICM function and then have amount_lose and amount_win be ICM values. In a first test, the convergence was starting and the results look promising.

Now the problem that I have is the exploitability calculation, as the solver will stop after much few iterations than with chipev and I can't seem to figure out how the computation works. It says:

/// Computes the expected values of the MES (Maximally Exploitative Strategy) of each player.
///
/// The bias, i.e., (starting pot) / 2, is already subtracted to increase the significant figures.
/// Therefore, the average of the return value corresponds to the exploitability value if not raked.
#[inline]
pub fn compute_mes_ev<T: Game>(game: &T) -> [f32; 2] {

But I can't find where this subtraction is happening. Admittedly I have never coded in Rust or C++ before so I apologize if I am missing something very obvious. How would I go to set the initial value for the exploitability?

**Comment by b-inary** [comment]
The subtraction is exactly what the `evaluate` function does, for example, see:
https://github.com/b-inary/postflop-solver/blob/7ba56e7263fb295a40b7296a556dc3a72b9eda73/src/game/evaluation.rs#L22-L26

So if you have already modified these `amount_win` and `amount_lose` properly, this should not be a problem. Rather, I suspect you need to modify the following conditional branch:
https://github.com/b-inary/postflop-solver/blob/7ba56e7263fb295a40b7296a556dc3a72b9eda73/src/utility.rs#L284-L290

Introducing ICM breaks the zero-sum property, so the `if` clause shortcut (L286) cannot be used. I am not confident, but I think the problem can be solved by simply using this `else` clause when calculating with the ICM.

**Comment by Barcode9** [comment]
Thanks that seemed to have stopped the convergence to 0!

**Comment by nastynaz** [comment]
@Barcode9 did you manage to implement ICM calculations? I'm currently working on it but not sure how the payoffs should be adjusted. I'm a fairly experienced rust dev with an academic background in game theory, but I'm not so knowledgeable about the poker theory of ICM payoffs. If you want to work together / help me out let me know, I'd greatly appreciate it.

**Comment by Barcode9** [comment]
Hi Naz,

Its been a while since I did this, I just wanted to see if I could implement ICM, the results looked reasonable so I figured it worked, but I didn't do extensive testing.

[evaluation.txt](https://github.com/user-attachments/files/16337953/evaluation.txt)

See the evaluation.rs file I changed. You also need to change what b-inary mentioned above. 

I think I read in the ICM mapping from a HoldemResources Node Export like this:

So you have a stack -> utility mapping and then I just did linear interpolation if I didn't have the exact mapping. 
[state.json](https://github.com/user-attachments/files/16337998/state.json)

Hope that helps!
